### PR TITLE
rwalk: init at 0.9.0

### PR DESCRIPTION
**Pull request by pwnwriter**
## Description of changes

A blazingly fast web directory scanner written in Rust

https://github.com/cestef/rwalk

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/rw/rwalk/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rwalk";
+  version = "0.8.7";
+
+  src = fetchFromGitHub {
+    owner = "cestef";
+    repo = "rwalk";
+    rev = "v${version}";
```
Comment (steeleduncan) [inline pkgs/by-name/rw/rwalk/package.nix:18]: `tag = "v${version}";`?

**Review by steeleduncan — CHANGES_REQUESTED** [review]

File: pkgs/by-name/rw/rwalk/package.nix (hunk 1 above)
Comment (pwnwriter) [reply pkgs/by-name/rw/rwalk/package.nix]: done

Comment (pwnwriter) [reply pkgs/by-name/rw/rwalk/package.nix]: idk why but the ci is showing error. I'll be working on that issue tomorrow if it's from my end.

Comment (steeleduncan) [reply pkgs/by-name/rw/rwalk/package.nix]: I think that now you have changed to a tag, the `{src.rev}` is no longer defined on line 49, and you'd need to use a tagged link to the release notes too

Comment (pwnwriter) [reply pkgs/by-name/rw/rwalk/package.nix]: Hell yeah, could you verify now?

Comment (steeleduncan) [reply pkgs/by-name/rw/rwalk/package.nix]: Ofborg runs again automatically once you've changed the PR. Should have the results within the day

Comment (steeleduncan) [reply pkgs/by-name/rw/rwalk/package.nix]: I think before it was failing on one of the GHA evals though, and that passed now, so it looks like it'll be fine

Comment (pwnwriter) [reply pkgs/by-name/rw/rwalk/package.nix]: Awesome 🎉

**Comment by steeleduncan** [comment]
@pwnwriter  Looks good to me, but I don't have commit access, and don't maintain a similar package, so it doesn't make sense for me to leave an approving review. Once the tests have passed, my suggestion to get it merged would be to find someone who has merged a rust tool comparable to this into `nixpkgs` recently, and ping them for a review

**Comment by pwnwriter** [comment]
@drupol, could you please check and merge ? Appreciate it :)

**Comment by drupol** [comment]
IIRC, embedding the `cargo.lock` file is not necessary any more. Have a look at https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/ty/typst/package.nix#L79

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by pwnwriter** [comment]
@GaetanLepage , I've committed the requested changes, But the ci fails now. Could you please guide me on this?

**Comment by drupol** [comment]
> @GaetanLepage , I've committed the requested changes, But the ci fails now. Could you please guide me on this? 

Run nixfmt-rfc-style on your nix file to fix the formatting

**Comment by pwnwriter** [comment]
should I now remove the lock file?

**Comment by pwnwriter** [comment]
@GaetanLepage Could you tell me how'd I retrieve the cargo hash? I thought it'll provide me if I try to build it. But yeah, I'm getting issue with `nix-update-script` not sure If it's intended. 

Screenshot 2024-12-30 at 5 45 54 PM

**Comment by pwnwriter** [comment]
The upstream repo doesn't provide Cargo lock file. I think I should first make a pr there and then will make changes here?

**Comment by pwnwriter** [comment]
The version has been bumped from 0.8.7 to 0.9.0. Should I discard this pr and create a new one as a follow up or this is fine as it is?

**Comment by pwnwriter** [comment]
done

**Review by GaetanLepage — APPROVED** [review]
LGTM, well done @pwnwriter :)

**Pull request merged**